src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -108,7 +113,15 @@ def get_ganesha_conf() -> str:
                 # fall back to default NFS port if not present in daemon_spec
                 "port": daemon_spec.ports[0] if daemon_spec.ports else 2049,
                 "bind_addr": bind_addr,
+                "haproxy_hosts": [],
             }
+            if spec.enable_haproxy_protocol:
+                # TODO: limit the list to IP on hosts running haproxy/ingress
+                context["haproxy_hosts"] = [
+                    self.mgr.inventory.get_addr(h)
+                    for h in self.mgr.inventory.keys()
+                ]
+                logger.debug("selected haproxy_hosts: %r", context["haproxy_hosts"])
```

Comment:
Yeah.... I'm in favor of keeping the debug log, but the TODO should probably be fixed. But I'm not sure how.
Can we get the narrow down the list of hosts here to only those this nfs service got scheduled on?  (actually the hosts the haproxy/ingress services got scheduled on but I think thats the same now)

Reply:
We'd only know the set of haproxy hosts after they've been deployed unfortunately. Although we have the same issue with the keepalived conf needing the IPs for all the hosts other than the ones its running on that will have haproxy daemons for that ingress service. There we handle it by just grabbing all the currently known daemons in the service and then also adding that set of daemons as dependencies for the keepalive daemon. So if the keepalive is deployed before all haproxy daemons are  deployed (meaning it has an incomplete list in its conf when deployed), once the final haproxy daemons are deployed, it will see that the dependencies have changed and reconfig the keepalive with the full set of host IPs it needs. In this particular case, we also just back out if no haproxy daemons are down yet since we know for sure a reconfig will have to happen anyway. It's not perfect since it still means sometimes we're going to have to deploy the keepalive and then reconfigure it just after, but that's the best we came up with for it so far. From the `keepalived_generate_config` in `mgr/cephadm/services/ingress.py`
```
        daemons = self.mgr.cache.get_daemons_by_service(spec.service_name())

        if not daemons and not spec.keepalive_only:
            raise OrchestratorError(
                f'Failed to generate  keepalived.conf: No daemons deployed for {spec.service_name()}')

        deps = sorted([d.name() for d in daemons if d.daemon_type == 'haproxy'])

        host = daemon_spec.host
        hosts = sorted(list(set([host] + [str(d.hostname) for d in daemons])))
```

So we'd probably need something like that, probably minus the check that backs out if no daemons are down. What will likely happen is it will deploy the nfs, deploy the ingress, then reconfig the nfs after once the ingress is deployed and it now knows the set of hosts.

Reply:
So, maybe I'm just being lazy but I've made my peace with the way the code is at the moment. It basically restricts NFS ganesha from taking proxy protocol values from hosts that are not managed by cephadm. We could do better by restricting it to only those hosts actually running keepalived but it may not be worth the extra effort. We have accomplished the goal of preventing ganesha from being "fronted" by a an arbitrary host talking the haproxy protocol. So unless we lose control of a cephadm managed node I think we could hold off on doing this until we have a good reason.
At the very least I'll change the comment from a "TODO" to a "NB" and if we decide that additonal complexity is worth it I'd ask to do it in a follow up pr. Hows that sound?

Reply:
I'm alright with that. I don't think having all the hosts in there would do any damage.

Reply:
Thanks. I think (hope?) that what we have now is good enough. I've updated the comment to reflect these conclusions. I think we're good to go wrt whatever final rounds of testing or discussion is needed before merging.